[CI] S3SnapshotRepoTestKitIT testRepositoryAnalysis failing

Turns out this has been tripping an assertion basically forever, but until https://github.com/elastic/elasticsearch/pull/102976 we were just swallowing it.

**Build scan:**
https://gradle-enterprise.elastic.co/s/fqyklzmrdoozc/tests/:x-pack:plugin:snapshot-repo-test-kit:qa:s3:javaRestTest/org.elasticsearch.repositories.blobstore.testkit.S3SnapshotRepoTestKitIT/testRepositoryAnalysis


**Reproduction line:**
```
./gradlew ':x-pack:plugin:snapshot-repo-test-kit:qa:s3:javaRestTest' --tests "org.elasticsearch.repositories.blobstore.testkit.S3SnapshotRepoTestKitIT.testRepositoryAnalysis" -Dtests.seed=AD352AD47F2A3AA5 -Dtests.locale=lt -Dtests.timezone=Europe/Oslo -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.repositories.blobstore.testkit.S3SnapshotRepoTestKitIT#testRepositoryAnalysis`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRepositoryAnalysis'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.repositories.blobstore.testkit.S3SnapshotRepoTestKitIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: parseRequestBody failed

  at __randomizedtesting.SeedInfo.seed([AD352AD47F2A3AA5]:0)
  at fixture.s3.S3HttpHandler.parseRequestBody(S3HttpHandler.java:390)
  at fixture.s3.S3HttpHandler.handle(S3HttpHandler.java:183)
  at fixture.s3.S3HttpFixture$1.handle(S3HttpFixture.java:58)
  at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)
  at sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)
  at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:101)
  at sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:871)
  at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)
  at sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:847)
  at sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:204)
  at sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:566)
  at sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:532)
  at java.lang.Thread.run(Thread.java:1583)

  Caused by: java.io.IOException: connection closed before all data received

    at sun.net.httpserver.FixedLengthInputStream.readImpl(FixedLengthInputStream.java:67)
    at sun.net.httpserver.LeftOverInputStream.read(LeftOverInputStream.java:99)
    at java.io.FilterInputStream.read(FilterInputStream.java:95)
    at org.elasticsearch.core.Streams.copy(Streams.java:45)
    at org.elasticsearch.core.Streams.copy(Streams.java:79)
    at org.elasticsearch.common.io.Streams.readFully(Streams.java:189)
    at fixture.s3.S3HttpHandler.parseRequestBody(S3HttpHandler.java:320)
    at fixture.s3.S3HttpHandler.handle(S3HttpHandler.java:183)
    at fixture.s3.S3HttpFixture$1.handle(S3HttpFixture.java:58)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)
    at sun.net.httpserver.AuthFilter.doFilter(AuthFilter.java:82)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:101)
    at sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(ServerImpl.java:871)
    at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:98)
    at sun.net.httpserver.ServerImpl$Exchange.run(ServerImpl.java:847)
    at sun.net.httpserver.ServerImpl$DefaultExecutor.execute(ServerImpl.java:204)
    at sun.net.httpserver.ServerImpl$Dispatcher.handle(ServerImpl.java:566)
    at sun.net.httpserver.ServerImpl$Dispatcher.run(ServerImpl.java:532)
    at java.lang.Thread.run(Thread.java:1583)

```

--- a/test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java
+++ b/test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java
@@ -21,6 +21,8 @@ import org.elasticsearch.common.regex.Regex;
 import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.SuppressForbidden;
 import org.elasticsearch.core.Tuple;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
 import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.rest.RestUtils;
 
@@ -52,6 +54,8 @@ import static org.w3c.dom.Node.ELEMENT_NODE;
  */
 @SuppressForbidden(reason = "this test uses a HttpServer to emulate an S3 endpoint")
 public class S3HttpHandler implements HttpHandler {
+
+    private static final Logger logger = LogManager.getLogger(S3HttpHandler.class);
 
     private final String bucket;
     private final String path;
@@ -382,12 +386,13 @@ public class S3HttpHandler implements HttpHandler {
             }
             return Tuple.tuple(MessageDigests.toHexString(MessageDigests.digest(bytesReference, MessageDigests.md5())), bytesReference);
         } catch (Exception e) {
+            logger.error("exception in parseRequestBody", e);
             exchange.sendResponseHeaders(500, 0);
             try (PrintStream printStream = new PrintStream(exchange.getResponseBody())) {
-                printStream.println(e.toString());
+                printStream.println(e);
                 e.printStackTrace(printStream);
             }
-            throw new AssertionError("parseRequestBody failed", e);
+            throw e;
         }
     }
 

--- a/x-pack/plugin/snapshot-repo-test-kit/qa/s3/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/S3SnapshotRepoTestKitIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/qa/s3/src/javaRestTest/java/org/elasticsearch/repositories/blobstore/testkit/S3SnapshotRepoTestKitIT.java
@@ -61,10 +61,4 @@ public class S3SnapshotRepoTestKitIT extends AbstractSnapshotRepoTestKitRestTest
 
         return Settings.builder().put("client", "repo_test_kit").put("bucket", bucket).put("base_path", basePath).build();
     }
-
-    @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/103012")
-    public void testRepositoryAnalysis() throws Exception {
-        super.testRepositoryAnalysis();
-    }
 }

